app/server.go

```diff
@@ -846,9 +848,13 @@ func (s *Server) Shutdown() {
 	}
 
 	// This must be done after the cluster is stopped.
-	if s.Jobs != nil && s.runjobs {
-		s.Jobs.StopWorkers()
-		s.Jobs.StopSchedulers()
+	if s.Jobs != nil {
+		if err = s.Jobs.StopWorkers(); err != nil {
+			mlog.Warn("Failed to stop job server workers", mlog.Err(err))
+		}
+		if err = s.Jobs.StopSchedulers(); err != nil {
```

Comment:
This does not check for the `RunScheduler` flag separately. There are four cases:
- only jobs, no schedulers
- only schedulers, no jobs
- both scheduler and jobs
- no scheduler, no jobs.

Wondering if this handles all four correctly?

Reply:
That code independently stops both jobs and schedulers (if running). Otherwise it's almost a no-op.
We could add checks of course but we are shutting down the server so I opted for keeping it simple and focus on the end result.

Reply:
Ah great. Can we add a comment to reflect the no-op part?